### ISSUE TITLE
[bitnami/redis-cluster]Scaled up node in redis-cluster does not rebalance/reshard

### ISSUE THREAD

**Issue by njaisw** [bitnami/charts#7579]
**Which chart**:
redis-cluster - https://harbor-repo.vmware.com/harbor/projects/71/helm-charts/redis-cluster/versions/6.2.3

**Describe the bug**
When you scale node from 6 nodes to 8 nodes, UpdateJob does not re-balance the master nodes.

**To Reproduce**
1: Deploy 6 node redis-cluster  (3 master + 3 slaves)
2: Scale up 8 nodes ( 4 master+ 3 slaves)
3: Log into one the deployed redis-cluster container to see the cluster state.
   kubectl exec redis-cluster -- redis-cli cluster nodes : <shows 4th node added has 0 hash slots>
   kubectl exec redis-cluster -- redis-cli cluster info : <cluster size still 3 as 4th master node does not have any hash slot assigned.
4: we have to log into container and manually rebalance the cluster by executing the command below.
 redis-cli --cluster rebalance 192.168.11.135:6379 --cluster-use-empty-masters

**Expected behavior**
The scaled up cluster must automatically rebalance the empty masters.

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
[root@localhost tcx]# helm version
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /root/.kube/tcx-nimbus-cluster-106
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /root/.kube/tcx-nimbus-cluster-106
version.BuildInfo{Version:"v3.5.4", GitCommit:"1b5edb69df3d3a08df77c9902dc17af864ff05d1", GitTreeState:"clean", GoVersion:"go1.15.11"}

```

- Output of `kubectl version`:

```
root@localhost tcx]# kubectl version
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.0", GitCommit:"cb303e613a121a29364f75cc67d3d580833a7479", GitTreeState:"clean", BuildDate:"2021-04-08T16:31:21Z", GoVersion:"go1.16.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.6", GitCommit:"d32e40e20d167e103faf894261614c5b45c44198", GitTreeState:"clean", BuildDate:"2020-05-20T13:08:34Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.21) and server (1.17) exceeds the supported minor version skew of +/-1

```

**Additional context**

**Comment by yilmi** [comment]
Hi @njaisw, could you please share the values you used and any other info that could help us reproduce it?

**Comment by njaisw** [comment]
1: Deployed 6 node cluster with default values, the cluster looked perfectly fine
**cluster:
    init: true
    nodes: 6
    replicas: 1
    update:
      addNodes: false
      currentNumberOfNodes: 6**

2: Scaled up 6 nodes to 8 nodes with values below
 **cluster:
    init: false
    nodes: 8
    replicas: 1
    update:
      addNodes: true
      currentNumberOfNodes: 6**

All 8 nodes came up but the scaled master has zero slot as you see below. We have to rebalance manually to use empty master and after resharding the cluster info shows proper slots distributed. Please note the more details below
     
[root@localhost ~]# kubectl get pods -o wide | grep redis-cl
redis-cluster-0                                          2/2     Running            0          67s     10.233.105.186   node4   <none>           <none>
redis-cluster-1                                          2/2     Running            0          115s    10.233.92.112    node3   <none>           <none>
redis-cluster-2                                          2/2     Running            0          2m16s   10.233.70.85     node5   <none>           <none>
redis-cluster-3                                          2/2     Running            0          2m37s   10.233.108.111   node6   <none>           <none>
redis-cluster-4                                          2/2     Running            0          2m59s   10.233.90.207    node1   <none>           <none>
redis-cluster-5                                          2/2     Running            0          3m10s   10.233.96.130    node2   <none>           <none>
**redis-cluster-6                                          2/2     Running            0          3m28s   10.233.108.110   node6   <none>           <none>
redis-cluster-7                                          2/2     Running            0          3m28s   10.233.96.129    node2   <none>           <none>**

[root@localhost ~]# kubectl exec redis-cluster-0 -- redis-cli cluster nodes
Defaulted container "redis-cluster" out of: redis-cluster, metrics, volume-permissions (init)
c3bcce5e9aee7076f9f83e8001fc2104f6fd0475 10.233.96.130:6379@16379 slave 487e7f379c62e4408bf0d23f0bb798d894cdce5a 1632411969435 1632411966394 2 connected
487e7f379c62e4408bf0d23f0bb798d894cdce5a 10.233.92.112:6379@16379 master - 0 1632411974476 2 connected 5461-10922
e410a8136c64a2aa32a5aa2d2bdb73d9433d75f9 10.233.70.85:6379@16379 master - 0 1632411969000 3 connected 10923-16383
287a5c28bb36d1d9973ef2817cb7c4866f0d30f2 10.233.108.111:6379@16379 slave e410a8136c64a2aa32a5aa2d2bdb73d9433d75f9 0 1632411972459 3 connected
497ec6ce8eecbd2bc89e779aedfa2cca0e6ba15e 10.233.90.207:6379@16379 slave 178f702ceac2efd900e411d4b44ea52c856a9bea 0 1632411968000 1 connected
**7914fe71c2da624f7ed2b9216aa0076332541f82 10.233.108.110:6379@16379 master - 0 1632411973469 0 connected**
178f702ceac2efd900e411d4b44ea52c856a9bea 10.233.105.186:6379@16379 myself,master - 0 1632411972000 1 connected 0-5460
8718d44a25b8476faaed4a83fb040d7d82260386 10.233.96.129:6379@16379 slave 7914fe71c2da624f7ed2b9216aa0076332541f82 0 1632411972991 0 connected
[root@localhost ~]# kubectl exec redis-cluster-0 -- redis-cli cluster info
Defaulted container "redis-cluster" out of: redis-cluster, metrics, volume-permissions (init)
cluster_state:ok
cluster_slots_assigned:16384
cluster_slots_ok:16384
cluster_slots_pfail:0
cluster_slots_fail:0
**cluster_known_nodes:8
cluster_size:3**
cluster_current_epoch:6
cluster_my_epoch:1
cluster_stats_messages_ping_sent:134
cluster_stats_messages_pong_sent:150
cluster_stats_messages_sent:284
cluster_stats_messages_ping_received:150
cluster_stats_messages_pong_received:134
cluster_stats_messages_fail_received:2
cluster_stats_messages_received:286
**[root@localhost ~]# kubectl exec redis-cluster-0 -- redis-cli --cluster rebalance 10.233.70.85:6379 --cluster-use-empty-masters**
Defaulted container "redis-cluster" out of: redis-cluster, metrics, volume-permissions (init)
>>> Performing Cluster Check (using node 10.233.70.85:6379)
[OK] All nodes agree about slots configuration.
>>> Check for open slots...
>>> Check slots coverage...
[OK] All 16384 slots covered.
>>> Rebalancing across 4 nodes. Total weight = 4.00
Moving 1366 slots from 10.233.92.112:6379 to 10.233.108.110:6379
##################################################################################################################################
Moving 1365 slots from 10.233.105.186:6379 to 10.233.108.110:6379
#####################################################
Moving 1365 slots from 10.233.70.85:6379 to 10.233.108.110:6379
#######################################################################################################################################################
[root@localhost ~]# kubectl exec redis-cluster-0 -- redis-cli cluster nodes
Defaulted container "redis-cluster" out of: redis-cluster, metrics, volume-permissions (init)
c3bcce5e9aee7076f9f83e8001fc2104f6fd0475 10.233.96.130:6379@16379 slave 487e7f379c62e4408bf0d23f0bb798d894cdce5a 0 1632412177716 2 connected
487e7f379c62e4408bf0d23f0bb798d894cdce5a 10.233.92.112:6379@16379 master - 0 1632412177000 2 connected 6827-10922
e410a8136c64a2aa32a5aa2d2bdb73d9433d75f9 10.233.70.85:6379@16379 master - 0 1632412180000 3 connected 12288-16383
287a5c28bb36d1d9973ef2817cb7c4866f0d30f2 10.233.108.111:6379@16379 slave e410a8136c64a2aa32a5aa2d2bdb73d9433d75f9 0 1632412177000 3 connected
497ec6ce8eecbd2bc89e779aedfa2cca0e6ba15e 10.233.90.207:6379@16379 slave 178f702ceac2efd900e411d4b44ea52c856a9bea 0 1632412178000 1 connected
**7914fe71c2da624f7ed2b9216aa0076332541f82 10.233.108.110:6379@16379 master - 0 1632412177000 7 connected 0-1364 5461-6826 10923-12287**
178f702ceac2efd900e411d4b44ea52c856a9bea 10.233.105.186:6379@16379 myself,master - 0 1632412178000 1 connected 1365-5460
8718d44a25b8476faaed4a83fb040d7d82260386 10.233.96.129:6379@16379 slave 7914fe71c2da624f7ed2b9216aa0076332541f82 0 1632412178722 7 connected
[root@localhost ~]# kubectl exec redis-cluster-0 -- redis-cli cluster info
Defaulted container "redis-cluster" out of: redis-cluster, metrics, volume-permissions (init)
cluster_state:ok
cluster_slots_assigned:16384
cluster_slots_ok:16384
cluster_slots_pfail:0
cluster_slots_fail:0
**cluster_known_nodes:8
cluster_size:4**
cluster_current_epoch:7
cluster_my_epoch:1
cluster_stats_messages_ping_sent:347
cluster_stats_messages_pong_sent:376
cluster_stats_messages_update_sent:1
cluster_stats_messages_sent:724
cluster_stats_messages_ping_received:376
cluster_stats_messages_pong_received:4443
cluster_stats_messages_fail_received:2
cluster_stats_messages_update_received:2
cluster_stats_messages_received:4823

**Comment by yilmi** [comment]
@njaisw I'll need a bit of time to reproduce this and get back to you. Let me know if you make any progress on your end.

**Comment by njaisw** [comment]
I think we need to just execute command "redis-cli --cluster rebalance 10.233.70.85:6379 --cluster-use-empty-masters" on one of the redis instance, once the cluster is scaled up, in the updateJob.

**Comment by yilmi** [comment]
Hi @njaisw apologies for the delay! Did you check the following documentation related to scaling up? - https://github.com/bitnami/charts/tree/master/bitnami/redis-cluster#adding-a-new-node-to-the-cluster

**Comment by njaisw** [comment]
@yilmi yes! You easily reproduce the issue by following the steps

**Comment by yilmi** [comment]
Hi @njaisw, thanks for the pointer! I was able to reproduce the issue you reported on my end and opened an internal tracker for it. 👍

**Comment by alemorcuq** [comment]
Fixed in #8619
